### ISSUE TITLE
falcon.testing.* is not (re-)exported

### ISSUE THREAD

**Issue by jap** [falconry/falcon#2387]
Now that falcon 4 with type hints has been released 🎉 , we've enabled type checking using mypy as well for the falcon namespace for our codebase.

Unfortunately, `falcon.testing.__init__` does not reexport the names it imports, which causes complaints by mypy, for example:
```
tests/test_subscription_data.py:13: error: Name "falcon.testing.TestClient" is not defined  [name-defined]
```

(PR will follow)

**Comment by CaselIT** [comment]
Hi,

We are using `__all__` to export the names from the inits, so the PR should use it instead of `as ...`

**Comment by CaselIT** [comment]
@vytas7 I don't remember if there was a script that generated the `__all__`

**Comment by jap** [comment]
I just did generated one manually (that is, my editor did most of the work)

**Comment by vytas7** [comment]
Resolved fixed in #2388.
